paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java

```diff
@@ -79,10 +82,15 @@ public void handleSplitRequest(int subtask, @Nullable String hostname) {
         // The following batch assignment operation is for two purposes:
         // To distribute splits evenly when batch reading to prevent a few tasks from reading all
         // the data (for example, the current resource can only schedule part of the tasks).
-        // TODO: assignment is already created in constructor, here can just assign per batch
-        List<FileStoreSourceSplit> splits = pendingSplitAssignment.remove(subtask);
```

Comment:
Can we make `pendingSplitAssignment` to `Map<Integer, Queue<FileStoreSourceSplit>>`?
And poll a batch from it.

Reply:
+1